pkg/auth/auth_config.go

```diff
@@ -0,0 +1,58 @@
+package auth
+
+import (
+	"github.com/kartverket/skiperator/api/v1alpha1/digdirator"
+	"github.com/kartverket/skiperator/api/v1alpha1/istiotypes"
+	"slices"
+)
+
+type AuthConfigs []AuthConfig
+
+type AuthConfig struct {
+	Spec         istiotypes.Authentication
+	Paths        []string
+	IgnorePaths  []string
+	ProviderURIs digdirator.DigdiratorURIs
+}
+
+func (authConfigs *AuthConfigs) GetIgnoredPaths() []string {
```

Comment:
```suggestion
func (authConfigs *AuthConfigs) GetIgnoredPaths() []string {
	var ignoredPaths map[string]string
	var allowPaths map[string]string
	if authConfigs != nil {
		for _, config := range *authConfigs {
			for _, ignoredPath := range config.IgnorePaths {
				ignoredPaths[ignoredPath] = ignoredPath
			}
			for _, allowPath := range config.Paths {
				allowPaths[allowPath] = allowPath
			}
		}

		for _, path := range allowPaths {
			if _, ok := ignoredPaths[path]; ok {
				delete(ignoredPaths, path)
			}

		}
	}
	return maps.Values(ignoredPaths)
}
```

refactored suggestion using maps instead